### PR TITLE
Uniformize documentation and cleanup some examples

### PR DESCRIPTION
**Pull request by AmishgaAlphonius**
# Description of the problem

- Lethe's documentation wasn't necessarily uniform.

# Description of the solution

- New rules were set to help the uniformity of the documentation and current documentation was updated.
  - [x] Capitalize titles and subtitles (even in the new visual maps)
  - [x] prm block-text indentation
  - [x] Partially fix references (parameters guide, incompressible flow examples and rpt examples)
  - [x] Uniformize titles and subsections in examples
 
- Review and update a few incompressible flow examples
  - [x] doc/source/examples/incompressible-flow/2d-lid‐driven-cavity-flow
  - [x] doc/source/examples/incompressible-flow/2d-flow-around-cylinder
  - [x] doc/source/examples/incompressible-flow/2d-taylor-couette-flow
  - [x] doc/source/examples/incompressible-flow/2d-naca0012-low-reynolds

# Documentation

- Pretty much all documentation files were touched.
- A new subsection named "General Rules and Format" was added to ``doc/source/contributing.rst``

# Future changes

- In example pages, ``Files Used in This Example`` and ``References`` subsections will have to be updated using the rules described in the "General Rules and Format" subsection of the Contributing page.

### CODE REVIEW

File: doc/source/structure.rst [hunk 1]
```diff
@@ -1,11 +1,11 @@
 ##########################
-Folder structure of Lethe
+Folder Structure of Lethe
 ##########################
 
 The folder structure of Lethe is relatively complex. Once you have cloned and compiled Lethe, it is normal that you might be confused as to the meaning of the various folders in the cloned directory (which we will refer to as the lethe folder) or in the build folder. This section of the documentation aims at demystifying these elements.
 
 ================
-The lethe folder
+The lethe Folder
```
Comment (lpsaavedra) [inline doc/source/structure.rst:8]:
```suggestion
The Lethe Folder
```

**Review by lpsaavedra — CHANGES_REQUESTED** [review]
Now the documentation is for sure more uniform. I have just a few comments but overall good job!

File: doc/source/examples/rpt/rpt.rst [hunk 2]
```diff
@@ -21,9 +21,9 @@ Radioactive Particle Tracking (RPT)
...
 
-      rpt_1 [label="Photon count calculation \nin a cylindrical vessel", href="https://lethe-cfd.github.io/lethe/examples/rpt/photon-count-calculation-in-a-cylindrical-vessel/photon-count-calculation-in-a-cylindrical-vessel.html", tooltip="Photon count calculation in a cylindrical vessel"]; 
+      rpt_1 [label="Photon Count Calculation \nin a Cylindrical Vessel", href="https://lethe-cfd.github.io/lethe/examples/rpt/photon-count-calculation-in-a-cylindrical-vessel/photon-count-calculation-in-a-cylindrical-vessel.html", tooltip="Photon count calculation in a cylindrical vessel"];
 
-      rpt_2 [label="Tuning count calculation model \nparameters with NOMAD", href="https://lethe-cfd.github.io/lethe/examples/rpt/tuning-parameters-with-nomad/tuning-parameters-with-nomad.html", tooltip="Tuning count calculation model parameters with NOMAD"]; 
+      rpt_2 [label="Tuning Count Calculation Model \nParameters with NOMAD", href="https://lethe-cfd.github.io/lethe/examples/rpt/tuning-parameters-with-nomad/tuning-parameters-with-nomad.html", tooltip="Tuning count calculation model parameters with NOMAD"];
```
[the remark this answers is not in the document]
Comment (AmishgaAlphonius) [reply doc/source/examples/rpt/rpt.rst:26]: It was supposed to be all caps, that my mistake, thanks for pointing it out! :)

File: doc/source/examples/rpt/tuning-parameters-with-nomad/tuning-parameters-with-nomad.rst [hunk 3]
```diff
@@ -1,34 +1,36 @@
 =======================================================
-Tuning count calculation model parameters with NOMAD
+Tuning Count Calculation Model Parameters with Nomad
```
Comment (AmishgaAlphonius) [inline doc/source/examples/rpt/tuning-parameters-with-nomad/tuning-parameters-with-nomad.rst:2]:
```suggestion
Tuning Count Calculation Model Parameters with NOMAD
```

File: doc/source/examples/incompressible-flow/2d-backward-facing-step/2d-backward-facing-step.rst [hunk 4]
```diff
@@ -292,17 +301,18 @@ For :math:`Re = 1000` :
...
+First, the results provided by *Lethe* are identical or so to all of the three selected studies for low Reynolds numbers (:math:`Re \leq 400`). After that point, both results form *Lethe* and from Erturk (2008) `[1] <https://doi.org/10.1016/j.compfluid.2007.09.003>`_ diverge from the experimental data of Armaly and al. (1983) `[2] <https://doi.org/10.1017/S0022112083002839>`_. According to `[1] <https://doi.org/10.1016/j.compfluid.2007.09.003>`_, this error is due to 3D effects that are more potent as the flow becomes more and more turbulent. Furthermore, there is also a less significant but clearly noticeable error between *Lethe* and Erturk (2008) `[1] <https://doi.org/10.1016/j.compfluid.2007.09.003>`_: the fact that certain tolerances have been set higher for higher Reynolds number cases might have underestimated the reattachment length. Also, first order elements have been used throughout the whole simulation process. Using second order elements for velocity, for instance, could yield better results for higher Reynolds numbers, however, at a higher computational cost. The following table illustrates the error at :math:`Re = 600` for first and second order velocity elements.
```
Comment (blaisb) [inline doc/source/examples/incompressible-flow/2d-backward-facing-step/2d-backward-facing-step.rst:315]:
```suggestion
First, the results provided by Lethe are identical or so to all of the three selected studies for low Reynolds numbers (:math:`Re \leq 400`). After that point, both results form *Lethe* and from Erturk (2008) `[1] <https://doi.org/10.1016/j.compfluid.2007.09.003>`_ diverge from the experimental data of Armaly and al. (1983) `[2] <https://doi.org/10.1017/S0022112083002839>`_. According to `[1] <https://doi.org/10.1016/j.compfluid.2007.09.003>`_, this error is due to 3D effects that are more potent as the flow becomes more and more turbulent. Furthermore, there is also a less significant but clearly noticeable error between *Lethe* and Erturk (2008) `[1] <https://doi.org/10.1016/j.compfluid.2007.09.003>`_: the fact that certain tolerances have been set higher for higher Reynolds number cases might have underestimated the reattachment length. Also, first order elements have been used throughout the whole simulation process. Using second order elements for velocity, for instance, could yield better results for higher Reynolds numbers, however, at a higher computational cost. The following table illustrates the error at :math:`Re = 600` for first and second order velocity elements.
```

File: doc/source/examples/incompressible-flow/2d-backward-facing-step/2d-backward-facing-step.rst [hunk 5]
```diff
@@ -292,17 +301,18 @@ For :math:`Re = 1000` :
...
-Validation and comparison
+-------------------------
+Validation and Comparison
 -------------------------
 
-Reattachment length
+Reattachment Length
 ~~~~~~~~~~~~~~~~~~~
 
 In this section, the solutions obtained with *Lethe* are compared with data that can be found in the scientific literature (Erturk (2008) `[1] <https://doi.org/10.1016/j.compfluid.2007.09.003>`_, Armaly and al. (1983) `[2] <https://doi.org/10.1017/S0022112083002839>`_ and Velivelli and Bryden (2015) `[3] <https://doi.org/10.1016/j.advengsoft.2014.11.006>`_). Several studies include datasets of :math:`x_r/h = f(Re)` (reattachment length) either experimentally or numerically. The next figure illustrates some of them in comparison with *Lethe*.
```
Comment (blaisb) [inline doc/source/examples/incompressible-flow/2d-backward-facing-step/2d-backward-facing-step.rst:311]:
```suggestion
In this section, the solutions obtained with Lethe are compared with data that can be found in the scientific literature (Erturk (2008) `[1] <https://doi.org/10.1016/j.compfluid.2007.09.003>`_, Armaly and al. (1983) `[2] <https://doi.org/10.1017/S0022112083002839>`_ and Velivelli and Bryden (2015) `[3] <https://doi.org/10.1016/j.advengsoft.2014.11.006>`_). Several studies include datasets of :math:`x_r/h = f(Re)` (reattachment length) either experimentally or numerically. The next figure illustrates some of them in comparison with *Lethe*.
```

File: doc/source/contributing.rst [hunk 6]
```diff
@@ -97,6 +97,49 @@ To build standalone HTML files like the CI would, enter the following commands:
...
+
+- Examples generally contain the following subsections in the listed order:
+
+  - **Features**: lists of features of the example.
+  - **Files Used in This Examples**: lists of files used in the example in alphabetical order.
+  - **Description of the Case**: describes the system studied in the example.
+  - **Parameter File**: describes the different parameter subsections involved in the example. Each subsection of the parameter file begins with its name as a sub-heading.
+  - **Running the Simulation**: displays the command used to run the example and gives an scale of the running duration.
+  - **Results** or **Results and Discussion**: displays results of the simulation and comments on them.
+  - **Possibility for Extension**: lists different interesting ways to extend the example.
```
Comment (blaisb) [inline doc/source/contributing.rst:136]: Can you add also a comment about the citation format used?

File: doc/source/examples/incompressible-flow/3d-nitsche-mixer-with-pbt-impeller/nitsche-mixer-with-pbt-impeller.rst [hunk 7]
```diff
@@ -12,29 +9,28 @@ This example illustrates how the transient flow in a stirred-tank can be simulat
 .. seealso::
 	This example is related to this article (to be published).
```
Comment (blaisb) [inline doc/source/examples/incompressible-flow/3d-nitsche-mixer-with-pbt-impeller/nitsche-mixer-with-pbt-impeller.rst:10]:
```suggestion
	This example is related to this article (to be published).
```
Add hyperlink to this article. It is published now.
Joachim, J., Daunais, C. A., Bibeau, V., Heltai, L., & Blais, B. (2023). A parallel and adaptative Nitsche immersed boundary method to simulate viscous mixing. Journal of Computational Physics, 488, 112189.

**Review by blaisb — APPROVED** [review]
Minor things to address. If you do them this PM this can be merged today

File: doc/source/examples/incompressible-flow/2d-taylor-couette-flow-nitsche/2d-taylor-couette-flow-nitsche.rst [hunk 8]
```diff
@@ -253,13 +260,14 @@ The mesh can be dynamically adapted using Kelly error estimates on the velocity,
     end
 
 
-Rest of the subsections
+Rest of the Subsections
 ~~~~~~~~~~~~~~~~~~~~~~~
 
-The non-linear and linear solvers subsections do not contain any new information in this example.
+The ``non-linear solver`` and ``linear solver`` subsections do not contain any new information in this example..
```
Comment (hepap) [inline doc/source/examples/incompressible-flow/2d-taylor-couette-flow-nitsche/2d-taylor-couette-flow-nitsche.rst:266]:
```suggestion
The ``non-linear solver`` and ``linear solver`` subsections do not contain any new information in this example.
```

File: doc/source/examples/incompressible-flow/2d-taylor-couette-flow/2d-taylor-couette-flow.rst [hunk 9]
```diff
@@ -1,26 +1,33 @@
...
 
-Description of the case
+
+-----------------------
+Description of the Case
 -----------------------
 
-The Taylor-Couette flow is the name of a fluid flow in the gap between two long concentric cylinders with different rotational velocities. One or both of these cylinders may rotate along the axis, however generally it is assumed that the outer cylinder is fixed, and the inner cylinder rotates with a constant angular velocity. For the Taylor-Couette flow, an analytical solution of the Navier-Stokes equations can be found, although this solution is not stable for all ranges of operating conditions and becomes unstable at high Reynolds number.
+Taylor-Couette flow is the name of a fluid flow in the gap between two long concentric cylinders with different rotational velocities. One or both of these cylinders may rotate along the axis, however generally it is assumed that the outer cylinder is fixed, and the inner cylinder rotates with a constant angular velocity. For the Taylor-Couette flow, an analytical solution of the Navier-Stokes equations can be found, although this solution is not stable for all ranges of operating conditions and becomes unstable at high Reynolds number.
```
Comment (hepap) [inline doc/source/examples/incompressible-flow/2d-taylor-couette-flow/2d-taylor-couette-flow.rst:30]:
```suggestion
Taylor-Couette flow is a fluid flow in the gap between two long concentric cylinders with different rotational velocities. One or both of these cylinders may rotate along the axis, however generally it is assumed that the outer cylinder is fixed, and the inner cylinder rotates with a constant angular velocity. For the Taylor-Couette flow, an analytical solution of the Navier-Stokes equations can be found, although this solution is not stable for all ranges of operating conditions and becomes unstable at high Reynolds number.
```

**Review by hepap — APPROVED** [review]
Great harmonization and special mention to your patience in changing 129 files :D

File: doc/source/contributing.rst (hunk 6 above)
Comment (AmishgaAlphonius) [reply doc/source/contributing.rst:136]: I did mention that it was the IEEE referencing style in the line below, but I could add a link to the guide and add an example for a journal article. What do you think?

File: doc/source/contributing.rst [hunk 10]
```diff
@@ -97,6 +97,49 @@ To build standalone HTML files like the CI would, enter the following commands:
 
 The generated files should be in the ``build/html`` directory. Open ``index.html`` in a browser to view the rendered documents.
 
+General Rules and Format
+------------------------
+
+Throughout the documentation, you may notice that the different pages follow a certain format to ensure uniformity of the documentation and help user's navigate more fluidly. Here are the different elements that must be considered when contributing to the documentation of Lethe:
+
+- Titles and subtitles must be capitalized following the *Chicago Manual of Style*:
```
[the remark this answers is not in the document]
Comment (AmishgaAlphonius) [reply doc/source/contributing.rst:105]: I can't find any clean, open source reference, but I'll put a link to the actual manual

**Review by lpsaavedra — APPROVED** [review]

File: doc/source/examples/multiphysics/dam-break/dam-break.rst [hunk 11]
```diff
@@ -158,18 +165,20 @@ documentation for more explanation on the phase filtration.
...
+
+In the ``initial conditions``, the initial velocity and initial position
 of the liquid phase are defined. The liquid phase is initially 
 defined as rectangle of length :math:`= 3.5` and height :math:`= 7`.
 
 .. code-block:: text
 
     #---------------------------------------------------
-    # Initial condition
+    # Initial Conditions
```
[the remark this answers is not in the document]
Comment (AmishgaAlphonius) [reply doc/source/examples/multiphysics/dam-break/dam-break.rst:181]: I'll do it in this one, it shouldn't take that long :)

**Pull request merged**